demisto_sdk/commands/error_code_info/error_code_info.py

```diff
@@ -36,7 +36,8 @@ def parse_function_parameters(sig: inspect.Signature):
             if get_origin(param.annotation) in [Union, Optional]:
                 param_type = get_args(param.annotation)[0]
 
-            if param_type is not str:
+            # param_type = str or param_type = "Optional[Type]" or param_type = "Union[Type1, Type2]"
+            if param_type is not str and not isinstance(param_type, str):
```

Comment:
can we use `not isinstance` instead of `is not str`?

Reply:
unfortunately not, because when the type of the argument in the error code's function is only `str` then `param_type` will evaluate to `str`, however in case where the type of the argument is `Optional[str]` for example it will evaluate to the string "Optional[str]", hence we need to handle both cases.

this is how the `annotation` method of param works unfortunately :(